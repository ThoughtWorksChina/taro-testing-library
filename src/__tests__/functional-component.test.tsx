import Taro, {useEffect, useState} from '@tarojs/taro';
import {Text} from "@tarojs/components";
import {StandardProps} from "@tarojs/components/types/common";
import { act, render } from '../index';

interface CounterProps extends StandardProps {
  initial?: number;
}

const Counter = (props: CounterProps) => {
  const [count, setCount] = useState(props.initial || 1)
  return (
    <Text
      onClick={() => {setCount(count+1)}}
      className="number custom-class"
    >{count}</Text>
  );
};

describe('functional component test', () => {
  it('should render component', () => {
    const { container } = render(<Counter />);
    const $number = container.querySelector('.number') as HTMLSpanElement;
    expect($number.innerHTML).toEqual('1');
  });

  it('should render component with props', () => {
    const initial = 10
    const { container } = render(<Counter initial={initial} />);
    const $number = container.querySelector('.number') as HTMLSpanElement;
    expect($number.innerHTML).toEqual(`${initial}`);
  });

  it('should render component with props', () => {
    const initial = 10
    const { container } = render(<Counter initial={initial} />);
    const $number = container.querySelector('.number') as HTMLSpanElement;
    expect($number.innerHTML).toEqual(`${initial}`);
  });

  it('should rerender when trigger setState hooks', () => {
    const { container } = render(<Counter />);
    const $number = container.querySelector('.number') as HTMLSpanElement;
    act(() => {
      $number.click()
    })
    expect($number.innerHTML).toEqual(`2`);
  });

  it('should rerender when excute rerender methods', () => {
    const { container, rerender } = render(<Counter />);
    expect(container.querySelector('.number').innerHTML).toEqual("1");
    rerender(<Counter initial={2} />)
    expect(container.querySelector('.number').innerHTML).toEqual("2");
  });
});
