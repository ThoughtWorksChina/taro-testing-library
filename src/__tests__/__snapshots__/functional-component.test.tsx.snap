// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functional component test should support snapshot 1`] = `"<div>component without state</div>"`;
